### ISSUE TITLE
Creating Pgcluster resource via manifest fails on v4.3.0

### ISSUE THREAD

**Issue by TochukwuVictor** [CrunchyData/postgres-operator#1530]
**Describe the bug**
After installing v4.3.0 of the postgres-operator, I am able to create a pgcluster using the pgo client, but not when using a kubernetes manifest. When using a manifest the operator gives the following error:
```
time="2020-05-14T20:04:54Z" level=debug msg="in createPVC" func="operator/pvc.Create()" file="operator/pvc/pvc.go:134" version=4.3.0
time="2020-05-14T20:04:54Z" level=debug msg="using dynamic PVC template" func="operator/pvc.Create()" file="operator/pvc/pvc.go:148" version=4.3.0
{
    "kind": "PersistentVolumeClaim",
    "apiVersion": "v1",
    "metadata": {
        "name": "",
        "labels": {
            "vendor": "crunchydata",
            "pgremove": "true",
            "pg-cluster": "failing-cluster"
        }
    },
    "spec": {
        "accessModes": [
            "ReadWriteOnce"
        ],
    "storageClassName": "openebs-zfspv",
        "resources": {
            "requests": {
                "storage": "10G"
            }
        }
    }
}
time="2020-05-14T20:04:54Z" level=debug msg="configmaps \"failing-cluster-config\" not found" func="kubeapi.GetConfigMap()" file="kubeapi/configmap.go:45" version=4.3.0
time="2020-05-14T20:04:54Z" level=debug msg="failing-cluster-config configmap not found " func="kubeapi.GetConfigMap()" file="kubeapi/configmap.go:46" version=4.3.0
time="2020-05-14T20:04:54Z" level=error msg="Unable to find configMap failing-cluster-config when attempting disable autofailover" func="util.ToggleAutoFailover()" file="util/failover.go:247" version=4.3.0
time="2020-05-14T20:04:54Z" level=error msg="error in pvc create: PersistentVolumeClaim \"\" is invalid: metadata.name: Required value: name or generateName is required" func="operator/pvc.CreateIfNotExists()" file="operator/pvc/pvc.go:98" version=4.3.0
time="2020-05-14T20:04:54Z" level=error msg="PersistentVolumeClaim \"\" is invalid: metadata.name: Required value: name or generateName is required" func="operator/cluster.AddClusterBase()" file="operator/cluster/cluster.go:76" version=4.3.0
time="2020-05-14T20:04:54Z" level=debug msg="publishing events.EventCreateClusterFailureFormat message Event Event CreateClusterFailure - ns [crunchy-test-copy] - user [admin] topics [[clustertopic]] timestamp [2020-05-14 20:04:54.714011973 +0000 UTC m=+256059.598815034] - (create cluster failure) clustername failing-cluster workflow  error PersistentVolumeClaim \"\" is invalid: metadata.name: Required value: name or generateName is required" func="events.Publish()" file="events/eventing.go:49" version=4.3.0
```

The issue is that the name for the PVC resource isn't getting created as evidenced by this line:
```
level=error msg="error in pvc create: PersistentVolumeClaim \"\" is invalid: metadata.name: Required value: name or generateName is required" func="operator/pvc.CreateIfNotExists()" file="operator/pvc/pvc.go:98" version=4.3.0
```
despite the fact that I'm using a manifest generated from a working pgcluster.

**To Reproduce**
1. Create pgcluster with pgo client: `pgo create cluster working-cluster -n crunchy-test --replica-count 1 --storage-config openebs-zfspv --pgbackrest-storage-type s3`
2. Get manifest of working pgcluster: `kubectl get pgcluster working-cluster -oyaml > failing-cluster.yaml`
3. Change the pgcluster name references in manifest and remove status lines so operator doesn't view cluster as processed already.
4. Deploy the new cluster by running `kubectl apply -f failing-cluster.yaml`
5. See PVC error in operator logs.

**Expected behavior**
In v4.2.x I was able to create pgclusters from manifests without issue. This should be the same on v4.3.x.

**Screenshots**
N/A

**Please tell us about your environment:**

* Operating System: **Ubuntu**
* Where is this running: **On-prem** 
* Storage being used (NFS, Hostpath, Gluster, etc): **openebs-zfspv**
* Container Image Tag: **Any**
* PostgreSQL Version: **Any**
* Platform (Docker, Kubernetes, OpenShift): **Kubernetes**
* Platform Version: 1.17.4

**Additional context**

<details>
<summary>Manifest of working pgcluster</summary>
<p>

```yaml
apiVersion: crunchydata.com/v1
kind: Pgcluster
metadata:
  labels:
    autofail: "true"
    backrest-storage-type: s3
    crunchy-pgbadger: "false"
    crunchy-pgha-scope: working-cluster
    crunchy_collect: "true"
    deployment-name: working-cluster
    name: working-cluster
    pg-cluster: working-cluster
    pg-pod-anti-affinity: ""
    pgo-backrest: "true"
    pgo-version: 4.3.0
    pgouser: admin
  name: working-cluster
  namespace: crunchy-test
spec:
  ArchiveStorage:
    accessmode: ""
    matchLabels: ""
    name: ""
    size: ""
    storageclass: ""
    storagetype: ""
    supplementalgroups: ""
  BackrestStorage:
    accessmode: ReadWriteOnce
    matchLabels: ""
    name: ""
    size: 10G
    storageclass: openebs-zfspv
    storagetype: dynamic
    supplementalgroups: ""
  PrimaryStorage:
    accessmode: ReadWriteOnce
    matchLabels: ""
    name: working-cluster
    size: 10G
    storageclass: openebs-zfspv
    storagetype: dynamic
    supplementalgroups: ""
  ReplicaStorage:
    accessmode: ReadWriteOnce
    matchLabels: ""
    name: ""
    size: 10G
    storageclass: openebs-zfspv
    storagetype: dynamic
    supplementalgroups: ""
  WALStorage:
    accessmode: ""
    matchLabels: ""
    name: ""
    size: ""
    storageclass: ""
    storagetype: ""
    supplementalgroups: ""
  backrestRepoPath: ""
  backrestResources:
    memory: 48Mi
  backrestS3Bucket: ""
  backrestS3Endpoint: ""
  backrestS3Region: ""
  ccpimage: crunchy-postgres-ha
  ccpimageprefix: registry.developers.crunchydata.com/crunchydata
  ccpimagetag: centos7-12.2-4.3.0
  clustername: working-cluster
  collectSecretName: working-cluster-collect-secret
  customconfig: ""
  database: working-cluster
  enableBackrestMemoryLimit: false
  enableMemoryLimit: false
  exporterport: "9187"
  name: working-cluster
  namespace: crunchy-test
  pgBouncer:
    enableMemoryLimit: false
    replicas: 0
    resources:
      memory: 24Mi
  pgbadgerport: "10000"
  pgoimageprefix: registry.developers.crunchydata.com/crunchydata
  podAntiAffinity:
    default: required
    pgBackRest: preferred
    pgBouncer: required
  policies: ""
  port: "5432"
  primaryhost: working-cluster
  primarysecretname: working-cluster-primaryuser-secret
  replicas: "1"
  resources:
    memory: 128Mi
  rootsecretname: working-cluster-postgres-secret
  secretfrom: ""
  shutdown: false
  standby: false
  status: completed
  syncReplication: null
  tablespaceMounts: {}
  tls:
    caSecret: ""
    tlsSecret: ""
  tlsOnly: false
  user: deployment-test
  userlabels:
    backrest-storage-type: s3
    crunchy_collect: "true"
    pg-pod-anti-affinity: ""
    pgo-version: 4.3.0
  usersecretname: working-cluster-deployment-test-secret
```

</p>
</details>

<details>
<summary>Manifest of failing pgcluster</summary>
<p>

```yaml
apiVersion: crunchydata.com/v1
kind: Pgcluster
metadata:
  labels:
    autofail: "true"
    backrest-storage-type: s3
    crunchy-pgbadger: "false"
    crunchy-pgha-scope: failing-cluster
    crunchy_collect: "true"
    deployment-name: failing-cluster
    name: failing-cluster
    pg-cluster: failing-cluster
    pg-pod-anti-affinity: ""
    pgo-backrest: "true"
    pgo-version: 4.3.0
    pgouser: admin
  name: failing-cluster
  namespace: crunchy-test
spec:
  ArchiveStorage:
    accessmode: ""
    matchLabels: ""
    name: ""
    size: ""
    storageclass: ""
    storagetype: ""
    supplementalgroups: ""
  BackrestStorage:
    accessmode: ReadWriteOnce
    matchLabels: ""
    name: ""
    size: 10G
    storageclass: openebs-zfspv
    storagetype: dynamic
    supplementalgroups: ""
  PrimaryStorage:
    accessmode: ReadWriteOnce
    matchLabels: ""
    name: failing-cluster
    size: 10G
    storageclass: openebs-zfspv
    storagetype: dynamic
    supplementalgroups: ""
  ReplicaStorage:
    accessmode: ReadWriteOnce
    matchLabels: ""
    name: ""
    size: 10G
    storageclass: openebs-zfspv
    storagetype: dynamic
    supplementalgroups: ""
  WALStorage:
    accessmode: ""
    matchLabels: ""
    name: ""
    size: ""
    storageclass: ""
    storagetype: ""
    supplementalgroups: ""
  backrestRepoPath: ""
  backrestResources:
    memory: 48Mi
  backrestS3Bucket: ""
  backrestS3Endpoint: ""
  backrestS3Region: ""
  ccpimage: crunchy-postgres-ha
  ccpimageprefix: registry.developers.crunchydata.com/crunchydata
  ccpimagetag: centos7-12.2-4.3.0
  clustername: failing-cluster
  collectSecretName: failing-cluster-collect-secret
  customconfig: ""
  database: failing-cluster
  enableBackrestMemoryLimit: false
  enableMemoryLimit: false
  exporterport: "9187"
  name: failing-cluster
  namespace: crunchy-test
  pgBouncer:
    enableMemoryLimit: false
    replicas: 0
    resources:
      memory: 24Mi
  pgbadgerport: "10000"
  pgoimageprefix: registry.developers.crunchydata.com/crunchydata
  podAntiAffinity:
    default: required
    pgBackRest: preferred
    pgBouncer: required
  policies: ""
  port: "5432"
  primaryhost: failing-cluster
  primarysecretname: failing-cluster-primaryuser-secret
  replicas: "1"
  resources:
    memory: 128Mi
  rootsecretname: failing-cluster-postgres-secret
  secretfrom: ""
  shutdown: false
  standby: false
  syncReplication: null
  tablespaceMounts: {}
  tls:
    caSecret: ""
    tlsSecret: ""
  tlsOnly: false
  user: deployment-test
  userlabels:
    backrest-storage-type: s3
    crunchy_collect: "true"
    pg-pod-anti-affinity: ""
    pgo-version: 4.3.0
  usersecretname: failing-cluster-deployment-test-secret
```

</p>
</details>

**Comment by jkatz** [comment]
Please see https://github.com/CrunchyData/postgres-operator/issues/1500#issuecomment-625775379

This is resolved in 4.3.1 which is being released within the next few days.

**Comment by TochukwuVictor** [comment]
@jkatz That workaround didn't solve my problem as it did for the user who opened the issue. Are you sure this is the same problem? Because I get the same PVC error even when using `STORAGE5_*` configs and my errors are not the same as referenced in the [description of the linked ticket](https://github.com/CrunchyData/postgres-operator/issues/1500#issue-614440529). That issue referenced an operator that was unable to deploy, which is something I didn't encounter.
